BubbleScan-AI/SheetProcessor.py

```diff
@@ -0,0 +1,23 @@
+from abc import ABC, abstractmethod
+import os
+import cv2
+import numpy as np
+
```

Comment:
The new SheetProcessor abstract class provides a base for various sheet types, reducing code duplication and making CustomProcessor and Scantron95945 easy to extend.